cfme/common/vm.py

```diff
@@ -651,6 +657,143 @@ def does_vm_exist_on_provider(self):
         """Check if VM exists on provider itself"""
         return self.provider.mgmt.does_vm_exist(self.name)
 
+    def is_suspended_on_provider(self):
+        """Check if VM is suspended via provider API."""
+        return self.provider.mgmt.can_suspend and self.provider.mgmt.is_vm_suspended(self.name)
+
+    def is_paused_on_provider(self):
+        """Check if VM is paused via provider API."""
+        return self.provider.mgmt.can_pause and self.provider.mgmt.is_vm_paused(self.name)
+
+    def is_running_on_provider(self):
+        """Check if VM is running via provider API."""
+        return self.provider.mgmt.is_vm_running(self.name)
+
+    def is_stopped_on_provider(self):
+        """Check if VM is stopped via provider API."""
+        return self.provider.mgmt.is_vm_stopped(self.name)
+
+    def start_on_provider(self):
+        """Start VM via provider API."""
+        return self.provider.mgmt.start_vm(self.name)
+
+    def stop_on_provider(self):
+        """Stop VM via provider API."""
+        return self.provider.mgmt.stop_vm(self.name)
+
+    def pause_on_provider(self):
+        """Pause VM via provider API (if supported)."""
+        if not self.provider.mgmt.can_pause:
+            raise TypeError("provider '{}' is unable to pause".format(self.provider.key))
+        return self.provider.mgmt.pause_vm(self.name)
+
+    def suspend_on_provider(self):
+        """Suspend VM via provider API (if supported)."""
+        if not self.provider.mgmt.can_suspend:
```

Comment:
Q: wouldn't be better to make sure that all providers which don't support such actions return NotImplementedError instead ?

Reply:
Just checked and wrapanapi (provider.mgmt) will actually raise that if we got to that point, so I will just get rid of the check/raise here.